token/services/vault/keys/keys.go

```diff
@@ -146,8 +136,25 @@ func CreateIssueActionMetadataKey(hash string) (string, error) {
 	return CreateCompositeKey(TokenKeyPrefix, []string{IssueActionMetadata, hash})
 }
 
-func CreateTransferActionMetadataKey(hash string) (string, error) {
-	return CreateCompositeKey(TokenKeyPrefix, []string{TransferActionMetadata, hash})
+func CreateTransferActionMetadataKey(txID string, subKey string, index uint64) (string, error) {
+	return CreateCompositeKey(TokenKeyPrefix, []string{TransferActionMetadata, txID, subKey, strconv.FormatUint(index, 10)})
+}
+
```

Comment:
It would be good to add a comment here describing what the method does.

Reply:
okay